benchmarks/bench_online_ocsvm.py

```diff
@@ -0,0 +1,290 @@
+        # normal data are those with attribute 2
+        # abnormal those with attribute 4
+        s = (y == 2) + (y == 4)
+        X = X[s, :]
+        y = y[s]
+        y = (y != 2).astype(int)
+
+    # Vectorizing data
+    if dataset_name == 'SF':
+        # Casting type of X (object) as string is needed for string categorical
+        # features to apply LabelBinarizer
+        lb = LabelBinarizer()
+        x1 = lb.fit_transform(X[:, 1].astype(str))
+        X = np.c_[X[:, :1], x1, X[:, 2:]]
+        y = (y != b'normal.').astype(int)
+
+    if dataset_name == 'SA':
+        lb = LabelBinarizer()
+        # Casting type of X (object) as string is needed for string categorical
+        # features to apply LabelBinarizer
+        x1 = lb.fit_transform(X[:, 1].astype(str))
+        x2 = lb.fit_transform(X[:, 2].astype(str))
+        x3 = lb.fit_transform(X[:, 3].astype(str))
+        X = np.c_[X[:, :1], x1, x2, x3, X[:, 4:]]
+        y = (y != b'normal.').astype(int)
+
+    if dataset_name in ['http', 'smtp']:
+        y = (y != b'normal.').astype(int)
+
+    print_outlier_ratio(y)
+
+    n_samples, n_features = np.shape(X)
+    if dataset_name == 'SA':  # LibSVM too long with n_samples // 2
+        n_samples_train = n_samples // 20
+    else:
+        n_samples_train = n_samples // 2
+
+    n_samples_test = n_samples - n_samples_train
+    print('n_train: ', n_samples_train)
+    print('n_features: ', n_features)
+
+    tpr_libsvm = np.zeros(n_axis)
+    tpr_online = np.zeros(n_axis)
+    fit_time_libsvm = 0
+    fit_time_online = 0
+    predict_time_libsvm = 0
+    predict_time_online = 0
+
+    X = X.astype(float)
+
+    gamma = 1 / n_features  # OCSVM default parameter
+
+    for random_state in random_states:
+
+        print(random_state)
+
+        X, y = shuffle(X, y, random_state=random_state)
+        X_train = X[:n_samples_train]
+        X_test = X[n_samples_train:]
+        y_train = y[:n_samples_train]
+        y_test = y[n_samples_train:]
+
+        if novelty_detection:
+            X_train = X_train[y_train == 0]
+            y_train = y_train[y_train == 0]
+
+        std = StandardScaler()
+
+        print('----------- LibSVM OCSVM ------------')
+        ocsvm = OneClassSVM(kernel='rbf', gamma=gamma, nu=nu)
+        pipe_libsvm = make_pipeline(std, ocsvm)
+
+        tstart = time()
+        pipe_libsvm.fit(X_train)
+        fit_time_libsvm += time() - tstart
+
+        tstart = time()
+        # scoring such that the lower, the more normal
+        scoring = -pipe_libsvm.decision_function(X_test)
+        predict_time_libsvm += time() - tstart
+        fpr_libsvm_, tpr_libsvm_, _ = roc_curve(y_test, scoring)
+
+        f_libsvm = interp1d(fpr_libsvm_, tpr_libsvm_)
+        tpr_libsvm += f_libsvm(x_axis)
+
+        print('----------- Online OCSVM ------------')
+        nystroem = Nystroem(gamma=gamma, random_state=random_state)
+        online_ocsvm = SGDOneClassSVM(nu=nu, random_state=random_state)
```

Comment:
You need to specify `tol` or `max_iter` to be future-proof and avoid a `FutureWarning`